enzyme/test/MLIR/ForwardMode/batched_scalar_merge.mlir

```diff
@@ -0,0 +1,27 @@
+// RUN: %eopt --enzyme-diff-batch %s | FileCheck %s
+
+module {
+  func.func @square(%x : f64) -> f64{
+    %y = arith.mulf %x, %x : f64
+    return %y : f64
+  }
+  func.func @dsq(%x : f64, %dx1 : f64, %dx2 : f64) -> (f64,f64) {
+    %r1 = enzyme.fwddiff @square(%x, %dx1) { activity=[#enzyme<activity enzyme_dup>], ret_activity=[#enzyme<activity enzyme_dupnoneed>]} : (f64, f64) -> (f64)
+    %r2 = enzyme.fwddiff @square(%x, %dx2) { activity=[#enzyme<activity enzyme_dup>], ret_activity=[#enzyme<activity enzyme_dupnoneed>] } : (f64, f64) -> (f64)
+    return %r1,%r2 : (f64,f64)
+  }
+}
+
+// CHECK:   func.func @dsq(%[[arg0:.+]]: f64, %[[arg1:.+]]: tensor<2xf64>) -> tensor<2xf64> {
```

Comment:
can we have a simpler test that just does the merge and not run AD?

Also we should but this in a different subfolder maybe Optimize (instead of ForwardMode)